### PR TITLE
add script for generating error constants

### LINKED ISSUE #32: Include error constants in includes in this repo?
Instead of expecting the libraries all to use explicit numeric constants for error codes, we can generate constants for them, and have include files in this repo; for example for go:

```
package ably

// ably error codes
const (
	ErrNothing                                                       = 10000
	ErrBadRequest                                                    = 40000
	ErrInvalidRequestBody                                            = 40001
	ErrInvalidParameterName                                          = 40002
	ErrInvalidParameterValue                                         = 40003
	ErrInvalidHeader                                                 = 40004
	ErrInvalidCredential                                             = 40005
	ErrInvalidConnectionID                                           = 40006
	ErrInvalidMessageID                                              = 40007
	ErrInvalidContentLength                                          = 40008
	ErrMaximumMessageLengthExceeded                                  = 40009
	ErrInvalidChannelName                                            = 40010
	...
	ErrMemberImplicitlyLeftPresenceChannelConnectionClosed           = 91100
)
```

This was generated by a script, and could easily also be done for any other language.

I propose we add this file at `<repo>/protocol/include/errors.go`, add the script somewhere, and add the other includes as and when we're updating the relevant libraries.

@mattheworiordan @SimonWoolf wdyt?

cc @gernest

### PR DESCRIPTION
**Pull request by gernest**
closes #32

### CODE REVIEW

File: scripts/errors-const/main.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	re := regexp.MustCompile(`\/\*([\s\S]*?)\*\/`)
+
+	// we remove the multiline comments
+	f = re.ReplaceAll(f, []byte{})
+	m := make(map[int]string)
+	err = json.Unmarshal(f, &m)
+	if err != nil {
+		log.Fatal(err)
+	}
+	var keys []int
```
Comment (ORBAT) [inline scripts/errors-const/main.go]: More descriptive variable names woulod be good. Examples/suggestions: `o` --> `opts`, `f` --> `outpFile`  `m` --> `codeToDesc`, `keys` --> `codes`, etc.

**Comment by gernest** [comment]
@ORBAT I think this is ready for review

**Comment by gernest** [comment]
@ORBAT  I will add `-url` flag to support downloading from url.

**Comment by ORBAT** [comment]
Closing this as per discussion on Slack.

**Pull request closed without merging**